lib/ui/platform_dispatcher.dart

```diff
@@ -1269,6 +1280,40 @@ class PlatformDispatcher {
 
   @Native<Handle Function()>(symbol: 'PlatformConfigurationNativeApi::DefaultRouteName')
   external static String _defaultRouteName();
+
+  /// Returns a [TextScaler] that scales font size according to the user's
+  /// platform preference.
+  ///
+  /// Many platforms allow users to scale text globally for better readability.
+  /// Given the font size the app developer specified, this [TextScaler]
+  /// converts it to the preferred font size that accounts for platform-wide
+  /// text scaling.
+  ///
+  /// The scaled value of the same font size input may change if the user changes
+  /// the text scaling preference (in system settings for example). The
+  /// [onTextScaleFactorChanged] callback can be used to monitor such changes.
+  ///
+  /// Instead of directly using this getter, applications should typically use
+  /// `MediaQuery.scaledFontSizeOf` instead to retrive the scaled font size in a
+  /// widget tree, so text resizes properly when the text scaling preference
+  /// changes.
+  ///
+  /// See also:
+  ///
+  ///  * [WidgetsBindingObserver], for a mechanism at the widgets layer to
+  ///    observe when this callback is invoked.
+  TextScaler get platformTextScaler {
+    final double textScaleFactor = _configuration.textScaleFactor;
+    if (textScaleFactor == 1) {
+      return TextScaler.noScaling;
+    }
+    return _platformTextScalerImplemented ? _SystemTextScaler(textScaleFactor) : TextScaler.linear(textScaleFactor);
+  }
+
+  // Returns a negative number when there's an error:
+  // -1 : GetScaledFontSize is not implemented on the current platform.
```

Comment:
Wondering if there's a way to make errors a little more structured to make this API a little more ergonomic and avoid magic numbers like -1.

Reply:
Removed error code -2. Is there a way to share the error code definition between java and dart? The -1 literal is used once in each language so would comments that explain the return value suffice?